admin/src/js/controllers/translation-languages.js

```diff
@@ -128,6 +128,10 @@ angular.module('controllers').controller('TranslationLanguagesCtrl',
       });
     };
 
+    $scope.$on('LanguagesUpdated', () => {
+      getLanguages();
+    });
+
```

Comment:
Using broadcast events is really heavy weight and requires changing everywhere that might update languages so it's overly coupled.

Instead I recommend always just listening for changes in the database and reacting to that. That way no matter how the language changes it'll be reflected in the UI. We have a `Changes` service that makes this easy, eg:

```js
const changeListener = Changes({
  key: 'update-languages', // some unique id
  filter: change => change.doc.type === 'translations',
  callback: () => getLanguages()
});
$scope.$on('$destroy', changeListener.unsubscribe);
```

Then the rest of the code goes away and we just follow the database for events.